apps/condo/domains/billing/utils/receiptQRCodeUtils.js

```diff
@@ -7,48 +8,68 @@ const { CONTEXT_FINISHED_STATUS: ACQUIRING_CONTEXT_FINISHED_STATUS } = require('
 const { PAYMENT_DONE_STATUS, PAYMENT_WITHDRAWN_STATUS } = require('@condo/domains/acquiring/constants/payment')
 const { CONTEXT_FINISHED_STATUS } = require('@condo/domains/miniapp/constants')
 
-const REQUIRED_QR_CODE_FIELDS = ['BIC', 'PayerAddress', 'PaymPeriod', 'Sum', 'PersAcc', 'PayeeINN', 'PersonalAcc']
+const REQUIRED_QR_CODE_FIELDS = ['BIC', 'PayerAddress', 'Sum', 'PersAcc', 'PayeeINN', 'PersonalAcc']
 
 /**
- * @typedef {Object} TQRCodeFields
- * @property {string} BIC
- * @property {string} PayerAddress
- * @property {string} PaymPeriod
- * @property {string} Sum
- * @property {string} PersAcc The resident's account
- * @property {string} PayeeINN
- * @property {string} PersonalAcc The bank account of the receiver
+ * Default day of month for detection of period. Before this date we use previous month, after - the next one
+ * This value is used as fallback for billingContext.settings.receiptUploadDate
+ * @type {number}
  */
+const DEFAULT_PERIODS_EDGE_DATE = 20
 
 /**
- * @param {string} qrStr The QR code string got from the picture
- * @return {TQRCodeFields}
+ * @param {TRUQRCodeFields} qrCode
+ * @param {string} fieldName
+ * @return {string}
  */
-function parseReceiptQRCode (qrStr) {
-    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(qrStr)
-
-    if (!matches) {
-        throw new Error('Invalid QR code')
-    }
+function getQRCodeField (qrCode, fieldName) {
+    const entries = Object.entries(qrCode)
+    const ientries = entries.map(([fieldName, value]) => [fieldName.toLowerCase(), value])
+    return get(Object.fromEntries(ientries), fieldName.toLowerCase())
```

Comment:
IMO it is too expensive operation to get field from object. Is there any way to simplify the way retrieve concrete field from the array?

Reply:
I'd like to search fields case-insentitively, but return fields as is. Usually qr-codes contain less than 10 fields.